### PR TITLE
[benchmarks/dynamo] Use small-output accuracy policy for SigLIP

### PR DESCRIPTION
**Pull request by warrendeng**
## Summary
The next Triton pin bump will include https://github.com/triton-lang/triton/pull/11189, which updates Triton’s bundled CUDA libdevice. This change is not included in PyTorch's current Triton pin. 

In particular, the Triton PR changes the implementation used for float32 `erf` (`__nv_erff`). On A10G, we observed that the updated `__nv_erff` changes GELU rounding in `vit_base_patch16_siglip_256`. This causes the benchmark’s synthetic training loss to fail its accuracy check:
  - Failing test: https://github.com/pytorch/pytorch/actions/runs/35206051520/job/105174646954
  - Eager RMSE against FP64: 0.13545
  - Compiled RMSE against FP64: 0.47106
  - Ratio: 3.48×
  - Current limit: 3×

We isolated the failure to the updated `__nv_erff`; when we restored only its previous implementation, the accuracy failure disappeared. We also verified that the updated implementation remains within 1 ULP of correctly rounded FP64 (while CUDA permits up to 2 ULP) so this is expected numerical variation from the new implementation.

The failure is limited to the synthetic scalar loss that the benchmark creates to run backward. Inference and all other training results still pass with the existing thresholds.

This PR adds SigLIP to the existing small-output policy similar to what has been previously done (https://github.com/pytorch/pytorch/pull/177042 for example) to allow a larger multiplier for small tensors.

## Test plan
  - Both affected A10G TIMM shards pass with this change:
    https://github.com/pytorch/pytorch/actions/runs/35167626498
  - `erff` accuracy sweep:
    https://github.com/pytorch/pytorch/actions/runs/34736991855
  - Nonscalar training results pass:
    https://github.com/pytorch/pytorch/actions/runs/34738810460

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @kadeng @chauhang @amjames @jataylo @azahed98

### CODE REVIEW

**Review by jansel — APPROVED** [review]

**Review by atalman — APPROVED** [review]
lgtm
